includes/class-sensei-grading.php

```diff
@@ -1131,6 +1131,33 @@ public static function get_graded_lessons_sum() {
 
 	}
 
+	/**
+	 * Get average grade of all lessons graded in all the courses.
+	 *
+	 * @since 4.2.0
+	 * @access public
+	 * @return double $graded_lesson_average_grade Average value of all the graded lessons in all the courses.
+	 */
+	public function get_graded_lessons_average_grade() {
+		// Fetching all the grades of all the lessons that are graded.
+		global $wpdb;
+		$comment_query_piece           = [];
+		$comment_query_piece['select'] = "SELECT SUM( {$wpdb->commentmeta}.meta_value ) AS grade_sum, COUNT( * ) as grade_count";
+		$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
+		$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade')";
+
+		$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.NoCaching -- Performance improvement.
+		$sum_result          = $wpdb->get_row( $comment_query );
+		$average_grade_value = 0;
+		if ( 0 === $sum_result->grade_count ) {
```

Comment:
I discovered after writing [a test](https://github.com/Automattic/sensei/pull/4904/files#diff-0e870d478e9be0a729348e4b7b107193d1d7422e27e13687a1edf806b1ba1ac1R194) for this that this can result in a division by 0 error because [`grade_count` is a `string`](https://github.com/Automattic/sensei/pull/4904/files#diff-15ca8a2e2992e136808c71f8edc63942547a9b2a1e9535f291e4fc27c3fbaf1fR1224). 😱

Reply:
Fixed the condition and added test like yours [here](https://github.com/Automattic/sensei/pull/4865/commits/9ad249ae5e6c4278ca6ee19dbe9e0044de324118)